backend/vnc_backend.pm

```diff
@@ -8,36 +8,37 @@ use feature qw/say/;
 use Data::Dumper qw(Dumper);
 use Carp qw(confess cluck carp croak);
 
-sub fetch_all_pending_screenshots() {
-    my ($self, $timeout, $min_time) = @_;
-
+sub request_screen_update($ ) {
+    my ($self) = @_;
     return unless $self->{'vnc'};
+    $self->{vnc}->send_update_request();
+}
 
-    eval {$self->{'vnc'}->update_framebuffer($timeout, $min_time);};
-
-    if ($@) {
-        bmwqemu::diag "VNC failed $@";
-        $self->close_pipes();
-    }
-
+sub capture_screenshot($ ) {
+    my ($self) = @_;
+    return unless $self->{'vnc'};
+    $self->{vnc}->fetch_pending_updates();
     return unless $self->{'vnc'}->_framebuffer;
```

Comment:
Must be a git fetch rebase push error on my side.  will check first thing
I'm at my desk.

On Thu, Mar 12, 2015 at 9:05 AM Stephan Kulow <notifications@github.com>
wrote:

> In backend/vnc_backend.pm
> <https://github.com/os-autoinst/os-autoinst/pull/209#discussion_r26284053>
> :
>
> >
> > -    eval {$self->{'vnc'}->update_framebuffer($timeout, $min_time);};
> > -
> > -    if ($@) {
> > -        bmwqemu::diag "VNC failed $@";
> > -        $self->close_pipes();
> > -    }
> > -
> > +sub capture_screenshot($ ) {
> > +    my ($self) = @_;
> > +    return unless $self->{'vnc'};
> > +    $self->{vnc}->fetch_pending_updates();
> >      return unless $self->{'vnc'}->_framebuffer;
>
> you said you tested, but this function doesn't seem to exist:
>
> DIE Can't locate object method "fetch_pending_updates" via package
> "backend::VNC" at /usr/lib/os-autoinst/backend/vnc_backend.pm line 20.
>
> (found in production stupid me)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/os-autoinst/os-autoinst/pull/209/files#r26284053>.
>